app/src/features/dataset_explorer/DatasetExplorer.tsx

```diff
@@ -45,9 +28,25 @@ function DatasetExplorer() {
   const [loadStatus, setLoadStatus] = useState<LoadStatus>("unloaded");
   const [previewedSourceId, setPreviewedSourceId] = useState("");
   const [data, setData] = useState([]);
+  const [datasets, setDatasets] = useState<DatasetMetadata[]>([]);
   const [columns, setColumns] = useState([]);
   const [activeFilter, setActiveFilter] = useState<Array<string>>([]);
 
+  /* This is called whenever the component is rendered */
+  useEffect(() => {
```

Comment:
I was going to comment about dataset loading strategies here but A) I don't want to block your PR, and B) this sent me a bit down the rabbit hole and I have a PR in progress to move the dataset loading logic into it's own place so it can be used across the entire app going forward. And also it fixes a race condition :P 

Anyway TL;DR this is fine as is, but I'll be sending you a PR after this to change it again

Reply:
SGTM!